### DISCUSSION TITLE
Reading Group Thread: The Simple Essence of Automatic Differentiation

### DISCUSSION THREAD

**Discussion by srush** [google-research/dex-lang#494] [General]
I plan to work my way through this paper `The Simple Essence of Automatic Differentiation` http://conal.net/papers/essence-of-ad/essence-of-ad-icfp.pdf over the next couple weeks. Feel free to chime in as well. 

Based on this thread: https://github.com/google-research/dex-lang/issues/454#issuecomment-770117681

(I basically have undergrad-level PL depth so it's going to be slow. But I'm going to try to summarize as I go.)

**Comment by oxinabox** [comment]
I will try to do the same.

**Comment by conal** [comment]
I recommend [this link](http://conal.net/papers/essence-of-ad/) instead, which will point you to abstract, talks (especially the Microsoft Research version), and related work.

Feel free to ask me questions along the way.

**Reply by breandan** [reply]
![image](https://user-images.githubusercontent.com/175716/106853579-f14e7100-6687-11eb-8eec-5d21204cbe04.png)

Hi Conal, in the paragraph highlighted above you mention the symmetry of the Hessian. I recently became aware that Hessians can be asymmetric if the second-order partials are not continuous. I was curious, are you only considering smooth functions or do you have an explanation for why [Schwarz' theorem](https://en.wikipedia.org/wiki/Symmetry_of_second_derivatives#Theorem_of_Schwarz) does not apply in your setting? Thank you!

**Reply by conal** [reply]
Hi Breandan. The point I was getting at in this paragraph is that multilinear maps are subsumed by curried linear maps. I don't think I have any insight on your question about Hessians and symmetry.
